### PR TITLE
fix(agw): Modified code at s1ap to remove dependency on common state manager class

### PR DESCRIPTION
**Pull request by rsarwad**
## Summary
fix(agw): Modified code at s1ap to remove dependency on common state manager class
In this PR, new class StateUtility is added that contains common state variables and function which shall be used across various tasks like s1ap, mme_app and spgw task
This PR partially addresses the issue, https://github.com/magma/magma/issues/11192

## Test Plan
Executed s1ap sanity test suite and unit test cases

### CODE REVIEW

**Review by nstng — APPROVED** [review]
bazel changes lgtm

File: lte/gateway/c/core/oai/include/state_utility.hpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+#include "lte/gateway/c/core/common/common_defs.h"
+#include "lte/gateway/c/core/oai/common/log.h"
+#ifdef __cplusplus
+}
+#endif
+
+#include "lte/gateway/c/core/oai/common/conversions.h"
+#include "lte/gateway/c/core/oai/common/redis_utils/redis_client.hpp"
+
+namespace {
```
[the remark this answers is not in the document]
Comment (ajahl) [reply lte/gateway/c/core/oai/include/state_utility.hpp]: @rsarwad  Can you fix it, please?

File: lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_state_converter.cpp [hunk 2]
```diff
@@ -129,15 +128,15 @@ TEST_F(S1APStateConverterTest, S1apStateConversionExpectedEnbCount) {
   enb_map.map = init_state->mutable_enbs();
   // Inserting 1 enb association
   enb_map.insert(enb_association.sctp_assoc_id(), enb_association);
-  // state_to_proto should update num_enbs to match expected eNB count on the
-  // map
+  // Write the state info to DB and should update num_enbs to match expected eNB
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_state_converter.cpp]:
```suggestion
  // Write the state info to DB and update num_enbs to match expected eNB
```

**Review by ssanadhya — APPROVED** [review]
LGTM, with minor comment

**Pull request merged**